pkg/providers/vsphere/internal/templates/factory.go

```diff
@@ -66,15 +68,29 @@ func (f *Factory) CreateIfMissing(ctx context.Context, datacenter string, machin
 	if err != nil {
```

Comment:
I followed the code paths through and it seems rather odd to be validating if an existing template has appropriately sized disks in a `CreateIfMissing` function. I don't think this really has any business validating existing templates. 

`CreateIfMissing` is called in [1 place only](https://github.com/aws/eks-anywhere/blob/24b3b939f727753d23f81f28cbbd7f7361f2a710/pkg/providers/vsphere/defaults.go#L95). I wonder if, right after calling `CreateIfMissing` we should validate the disk. 

We can optimize further by changing `CreateIfMissing` to just `Create` and return an error that can be inspected. E.g. `ErrTemplateExists` and `errors.Is(err, templates.ErrTemplateExists)`. With that error, we can choose to ignore it and then validate the 'existing' template. 

These changes would make the code far more explicit in its intent.